### PR TITLE
Fix keys alignment of Select (One Ofs) elements

### PR DESCRIPTION
**Pull request by Baggz**

### CODE REVIEW

**Comment by XVincentX** [comment]
@Baggz Please attach the trello card

**Comment by Baggz** [comment]
@XVincentX 

https://trello.com/c/Z3lzNLg3/243-fix-stairway-to-heaven-effect

File: src/Components/ObjectProperties/ObjectProperties.js [hunk 1]
```diff
@@ -46,12 +46,18 @@ class ObjectProperties extends React.Component {
     super(props);
 
     this.state = {
-      keyWidth: null,
+      keyWidth: this.props.keyWidth || null,
```
Comment (XVincentX) [inline src/Components/ObjectProperties/ObjectProperties.js]: Why are you saving this into the state? I do not see any usage.

File: src/Components/ObjectProperties/ObjectProperties.js [hunk 2]
```diff
@@ -46,12 +46,18 @@ class ObjectProperties extends React.Component {
...
     };
 
     this.keyWidthsIndex = {};
   };
 
+  componentWillReceiveProps = (nextProps) => {
+    if (nextProps.keyWidth) {
+      this.setState({keyWidth: nextProps.keyWidth});
```
Comment (XVincentX) [inline src/Components/ObjectProperties/ObjectProperties.js]: I am afraid this is an anti pattern. You might potentially trigger 2 re-renders with this code. If you want to set an initial state, do that from the `constructor` function.

File: src/Components/ObjectProperties/ObjectProperties.js [hunk 3]
```diff
@@ -57,6 +59,12 @@ class ObjectProperties extends React.Component {
     this.subscription = this.context.eventEmitter.addListener('alignKeys', this.alignKeys);
   };
 
+  componentWillReceiveProps = (nextProps) => {
+    if (nextProps.keyWidth) {
+      this.setState({ keyWidth: nextProps.keyWidth });
```
Comment (XVincentX) [inline src/Components/ObjectProperties/ObjectProperties.js]: I am afraid this is an anti pattern - you could potentially trigger the rendering function twice. What's the aim here?

File: src/Components/ObjectProperties/ObjectProperties.js (hunk 1 above)
Comment (Baggz) [inline src/Components/ObjectProperties/ObjectProperties.js]: @XVincentX 

Please have a look at https://github.com/apiaryio/attributes-kit/blob/267717fb7b42a71fcda5eb5465506c0d85463229/src/Components/ObjectProperties/ObjectProperties.js#L70-L111.

File: src/Components/ObjectProperties/ObjectProperties.js (hunk 2 above)
Comment (Baggz) [inline src/Components/ObjectProperties/ObjectProperties.js]: @XVincentX 

But this is called during the lifecycle of the Attributes Kit, not with the initial rendering of AK.

File: src/Components/ObjectProperties/ObjectProperties.js (hunk 3 above)
Comment (Baggz) [inline src/Components/ObjectProperties/ObjectProperties.js]: Answered here—https://github.com/apiaryio/attributes-kit/pull/441#discussion_r91507066.

Comment (XVincentX) [inline src/Components/ObjectProperties/ObjectProperties.js]: Ok I verified on React documentation - it seems perfectly fine to call `setState` in this lifecycle method.

File: src/Components/ObjectProperties/ObjectProperties.js (hunk 1 above)
Comment (XVincentX) [inline src/Components/ObjectProperties/ObjectProperties.js]: 👍

**Comment by XVincentX** [comment]
@Baggz Last thing: Could you provide a test MSON structure for that?

**Pull request merged**